internal/rty/layouts.go

```diff
@@ -124,8 +124,9 @@ type concatLayoutComponent struct {
 }
 
 type ConcatLayout struct {
```

Comment:
i'm concerned that this component is already very complex. The semantics are poorly defined (is it a grid? is it a flexbox?). It has historically proven to be very buggy.

This change increases the complexity a lot. I'm certain there are configurations that have bugs.

i think you can revert this whole file, and replace it with a very simple change: if there are no edits, add a dynamic filler string. I think that would solve your problem without having to introduce AddRight. If that doesn't work, let's talk about this in person.

Reply:
just to capture a bit of offline discussion:
the problem is that what we want is, e.g.:
"TILT: OK .  EDITS: main.go, blah.go (+3 more) .                  DURATION: 3s"

We want "DURATION: 3s" to be right-aligned, and everything else to be left-aligned.
We want the file list ("main.go, blah.go") to be the first thing truncated when we're short on space.
That is, we want:
left: component component truncatable-component component
right: component

It's not clear what general layout concept would support this without getting overly complex, so for now I've just added an EditStatusLine that does what we need.